File: Sliding Window/1522번 - 문자열 교환/1522.cpp
#include <iostream>
#include <string>

using namespace std;

string str = "";
int ret = 1000;

void input() {
    cin >> str;
}

void solve() {
    int a_cnt = 0;
    int str_len = str.length();
    
    for(char a : str) {
        if(a == 'a') a_cnt += 1;
    }
    
    for(int i = 0; i < str_len; i++) {
        int b_cnt = 0;
        for(int j = 0; j < a_cnt; j++) {
            if(str[(i + j) % str_len] == 'b') b_cnt += 1;
        }
        ret = min(ret, b_cnt);
    }
    
    cout << ret << '\n';
}


int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    input();
    solve();
    
    return 0;
}

File: Sliding Window/1522번 - 문자열 교환/1522.py
import sys

input = sys.stdin.readline

st = input().rstrip()

def solve():
    a_cnt, ans = 0, 1000
    for c in st:
        if c == 'a':
            a_cnt += 1
    
    l = len(st)
    for i in range(l):
        b_cnt = 0
        for j in range(a_cnt):
            if st[(i + j) % l] == 'b':
                b_cnt += 1
        ans = min(ans, b_cnt)
    
    print(ans)

if __name__ == '__main__':
    solve()